### PR TITLE
Fix sentence format for picky parsers like Isailor/Android

### PR DESCRIPTION
**Pull request by netAction**

### CODE REVIEW

**Comment by joabakk** [comment]
I don't think it is correct to send an assumed magnetic variation of 0.0. Rather it should skip this value (`,,`)

**Comment by netAction** [comment]
You are right. This will be fine:

      day+month+year,
      '', ''
    ]);

File: index.js [hunk 1]
```diff
@@ -169,17 +169,25 @@ module.exports = function(app) {
         var hours = ('00' + datetime.getHours()).slice(-2);
         var minutes = ('00' + datetime.getMinutes()).slice(-2);
         var seconds = ('00' + datetime.getSeconds()).slice(-2);
+
+        var day = ('00' +datetime.getUTCDate()).slice(-2);
+        var month = ('00' +(datetime.getUTCMonth() + 1)).slice(-2); //months from 1-12
+        var year = ('00' +datetime.getUTCFullYear()).slice(-2);
+
         return toSentence([
-          '$SKRMC', hours + minutes + seconds + '.020',
```
Comment (sbender9) [inline index.js]: I'm not sure about taking this out. It must have been there for a reason. Anyone know why? 

I also noticed that the decoder @ http://freenmea.net/decoder has the decimal point when you click "Get Random NMEA"

Comment (netAction) [reply index.js]: Yes, it is a decimal point of seconds. The given timestamp format does not have miliseconds.

**Comment by tkurki** [comment]
Sorry about this. I think the original RMC was my doing, more a quick hack than nothing tested and usable, but ended up in the published plugin anyway.

Can we add `navigation.magneticVariation` with a default value of 0?. The code does not yet support setting defaults, but maybe it's time to add a `defaults: [.....]` as a sibling to keys. 

I think it could work so that if the encoder has `defaults` any keys whose default is not `undefined` the stream is replaced with `original.merge(Bacon.once(default))`.

If there is no default, no variation available and you just add `navigation.magneticVariation` as an extra input parameter the combineWith will never return anything.

**Comment by tkurki** [comment]
And I think the `.020` i copied for some strange reason from some other code, can't remember.

**Comment by netAction** [comment]
So this might fix the missing values. I think it is nice to add missing values to make the function feature complete, but more important the format should be correct. So the missing magnetic data is a slightly different issue.


    keys: [
      'navigation.datetime', 'navigation.speedOverGround', 'navigation.courseOverGroundTrue', 'navigation.position', 'navigation.magneticVariation',
    ],
    f: function(datetime8601, sog, cog, position, magneticVariation) {

...

    Math.abs(magneticVariation * 1).toFixed(2),
    magneticVariation < 0 ? 'W' : 'E',

**Comment by tkurki** [comment]
If you fix it that way and there is no variation it will never output anything. Not sure if that is relevant, just thought that sensible defaults (empty?) for all the values would be prudent.

But as you said that is another issue and can be handled separately.

Is this ok without the .020?

File: index.js [hunk 2]
```diff
@@ -140,8 +140,8 @@ module.exports = function(app) {
       This is one of the sentences commonly emitted by GPS units.
 
       12
```
Comment (tkurki) [inline index.js]: Could you please fix the column position for 12 as well?

**Comment by tkurki** [comment]
Oops my mistake, you already removed the .020.

**Pull request merged**